src/hotspot/share/prims/jvmtiAgent.cpp

```diff
@@ -504,6 +507,15 @@ static bool invoke_Agent_OnAttach(JvmtiAgent* agent, outputStream* st) {
     agent->set_os_lib(library);
     agent->set_loaded();
   }
+
+  // Print warning if EnableDynamicAgentLoading not enabled on the command line
+  if (!FLAG_IS_CMDLINE(EnableDynamicAgentLoading) && !agent->is_instrument_lib() && !JvmtiAgentList::is_loaded(library)) {
```

Comment:
While looking at some code related to this, I noticed a couple of typos in the pre-existing load_agent_from_executable() comment. See lines 265 ("cant't") and 268 (".&&"). Maybe you could clean them up.

Reply:
Nit: I'd suggest to add dot at the end of comment at 511.